docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -957,6 +957,45 @@ public class Processor {
 }
 ----
 
+=== Ignoring Class-Level Interceptors for Methods
```

Comment:
`Class-Level Interceptors` -> `Class-Level Interceptor Bindings` ... nitpicking I know but I think that we should be accurate here.

Reply:
I think both are actually fine and in the end mean the same thing. It doesn't really matter if the thing we ignore is the interceptor binding or the interceptor itself. That said, I don't feel strongly about this (unlike the annotation name :-) ), so fixed.

Reply:
> It doesn't really matter if the thing we ignore is the interceptor binding or the interceptor itself.

Well, it does matter because we ignore the bindings and not the interceptor. In fact, there is no such thing like a class-level interceptor. For example, you can use a class-level binding and method-level binding of the same annotation type and the method-level binding overrides the class-level binding. Now if you add `@NoInterceptor` the interceptor is not ignored, just the binding. I know it's hypothetical but let's be accurate to avoid confusion ;-)

Reply:
I hear you, but keep in mind you're being more precise than the Interceptors specification itself :-) See e.g. https://jakarta.ee/specifications/interceptors/2.0/interceptors-spec-2.0.html#excluding_interceptors or https://jakarta.ee/specifications/interceptors/2.0/apidocs/jakarta/interceptor/excludeclassinterceptors

Reply:
Sure, but these sections are about EJB-style `@Interceptors` which are not supported by ArC anyway...

Reply:
It's just a different way of how interceptors are associated with targets :-) I also noticed that the `@ExcludeClassInterceptors` javadoc says "Use of this annotation to exclude interceptors defined by means of interceptor binding annotations is not portable.", which suggests that implementations might support it, so perhaps not that terrible idea after all...

Reply:
> It's just a different way of how interceptors are associated with targets

Sure, but not idiomatic for CDI.

> "Use of this annotation to exclude interceptors defined by means of interceptor binding annotations is not portable."
> so perhaps not that terrible idea after all...

What exactly? Like to reuse this annotation for class-level bindings?

Reply:
> > It's just a different way of how interceptors are associated with targets
> 
> Sure, but not idiomatic for CDI.

The terminology still applies though. That said, I changed this heading to what you suggested, so I'd say we should move on :-)

> > "Use of this annotation to exclude interceptors defined by means of interceptor binding annotations is not portable."
> > so perhaps not that terrible idea after all...
> 
> What exactly? Like to reuse this annotation for class-level bindings?

Yes. I mean, I'd still be in favor of our own annotation, but that mention in the javadoc made me think that using `@ExcludeClassInterceptors` wouldn't be that bad.